std_workflow_dev/js/std_workflow_debug.js

```diff
@@ -0,0 +1,131 @@
+                    } else { uid = currentlyWith.id; }
+
+                    builder.panel(1).style("special").element(1, {
+                        deferred: P.template("debug/quick-actions").deferredRender({
+                            M: this,
+                            uid: (uid !== O.currentUser.id ? uid : undefined)
+                        })
+                    });
+
+                    var debugEntities;
+                    var entities = this.entities;
+                    if(entities) {
+                        debugEntities = [];
+                        var usedAsActionableBy = {};
+                        _.each(this.$states, function(defn,name) {
+                            if(defn.actionableBy) {
+                                usedAsActionableBy[defn.actionableBy] = true;
+                            }
+                        });
+                        _.each(entities.$entityDefinitions, function(v,name) {
+                            var first = entities[name+'_refMaybe'];
+                            var user = first ? O.user(first) : undefined;
+                            var i = {
+                                entity: name,
+                                uid: user ? user.id : undefined,
+                                personName: user ? user.name : undefined,
+                                usedAsActionableBy: usedAsActionableBy[name]
+                            };
+                            if(user) { debugEntities.unshift(i); } else { debugEntities.push(i); }
+                        });
+
+                        builder.panel(1).style("special").element(1, {
+                            deferred: P.template("debug/sidebar").deferredRender({
+                                M: this,
+                                debugEntities: debugEntities
+                            })
+                        });
+                    }
+
+                    adminPanel.link(99, "/do/workflow-dev/debug/debug-mode/disable/"+this.workUnit.id, "DISABLE DEBUG MODE", "terminal");
+                }
+            });
+        });
+
+        // --------------------------------------------------------------------------
+
+        P.respond("GET,POST", "/do/workflow-dev/debug/transition-to-previous-state", [
+            {pathElement:0, as:"workUnit", allUsers:true} // Security check below
+        ], function(E, workUnit) {
+            var M = getCheckedInstanceForDebugging(workUnit);
+            var select = M.$timeline.select().where("workUnitId","=",M.workUnit.id).order("id",true).limit(2);
+            var row = select.length > 1 ? select[1] : undefined;
+            if(!row) { O.stop(); }
+            M._forceMoveToStateFromTimelineEntry(row, null);
+            return E.response.redirect(M.url);
+        });
+
+        // --------------------------------------------------------------------------
+
+        P.respond("GET,POST", "/do/workflow-dev/debug/debug-mode", [
+            {pathElement:0, as:"string"},
```

Comment:
Isn't this redundant now?

Reply:
Can be used as non-auth user is the only main difference currently I think, *but* I want to/am imagining that the two diverge, where debug has more power and abilities that wouldn't be suitable for administrators even in the production system.